app/assets/src/components/Samples.jsx

```diff
@@ -626,18 +638,22 @@ class Samples extends React.Component {
       .then(res => {
         this.setState(prevState => ({
           isRequesting: false,
-          allSamples: [...prevState.allSamples, ...res.data.samples],
+          fetchedSamples: merge(
```

Comment:
from the lodash conversation: couldn't this be just `Object.assign`?

Reply:
It's true. We can do `Object.assign({}, a, b)` or `{...a, ...b}` for immutability. Some of the other `lodash` methods are harder to replace. I would prefer to keep `lodash` to speed up development time unless bundle size is a big issue, in which case we might consider writing our own small set of utility functions.